src/providers/extraction_providers/extractionProviderFactory.ts

```diff
@@ -0,0 +1,14 @@
+import { ExtractionProvider } from './extractionProvider';
+import { MockExtractionProvider } from './mockExtractionProvider';
+
+interface ExtractionConfig {
+	extractionProviderName: 'mock';
+}
```

Comment:
sorry again, we should follow @blam2000's style with enums:

```suggestion
export const enum ExtractionProviderName {
	mock = 'MOCK';
}

export interface ExtractionConfig {
	extractionProviderName: string;
}
```